rollup-node.md

```diff
@@ -0,0 +1,285 @@
+- `random` is currently set to all zero bytes. After [the merge][merge], the randomness
+  value exposed for the L1 block will be copied in this property.
+- `suggestedFeeRecipient` is set to an address where the sequencer would like to
+  direct the fees
+- `transactions` is an array of transactions, RLP-encoded in the [EIP-2718]
+  format (i.e. as a sequence of a single byte and a byte array), of the allowed
+  types. We specify how to select the transactions below.
+
+[unix type]: https://en.wikipedia.org/wiki/Unix_time
+[merge]: https://ethereum.org/en/eth2/merge/
+[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+
+> For details on RLP encoding, refer to the yellow paper, or look at [this
+> hyperlinked implementation][encode-tx].
+
+[encode-tx]: https://github.com/norswap/nanoeth/blob/cc5d94a349c90627024f3cd629a2d830008fec72/src/com/norswap/nanoeth/transactions/Transaction.java#L84-L130
+
+In particular, we allow the following two transaction types:
+- [EIP-1559]: the current "regular" Ethereum L1 transaction type, which
+  separates gas fees between a base fee and a tip
+- the *[L1 attributes transaction]*, see below
+
+[EIP-1559]: https://eips.ethereum.org/EIPS/eip-1559
+
+Historical Optimistic Ethereum transactions also include [EIP-155] transactions,
+which are not accepted by the network anymore. This is not an [EIP-2718] typed
+transaction, but can be distinguished from those by the fact that their
+RLP-encoding's first byte will always be `>= 0xC0`.
+
+[EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+
+#### Payload Transaction Format
+
+The `transactions` array is filled with the deposits, prefixed by the (single)
+[L1 attributes transaction]. The deposits (which must be similarly RLP-encoded)
+are simply copied byte-for-byte — it is the role of the [execution engine] to
+reject invalidly-formatted transactions.
+
+> **TODO** must offer some precisions on the format of deposits: sender,
+> receivers both in-tx-as-encoded, and on-L2-tx. What about the fees?
+
+The Optimistic Ethereum specific *[L1 attributes transaction]* has the following
+[EIP-2718]-compatible format: `0x7E || [block_number, timestamp, basefee]`
+where:
+
+- `0x7E` is the transaction type identifier. It is selected because transaction
+  type identifiers are currently allowed to go up to `0x7F`. Picking a high
+  identifier minimizes the risk that the identifier will be used by Ethereum in
+  the future. We don't pick `0x7F` itself in case it becomes used for a
+  variable-length encoding scheme.
+- `block_number` is the L1 block number as a 64-bit integer (4 bytes)
+- `timestamp` is the L1 block timestamp as a 64-bit integer (4 bytes)
+- `basefee` is the L1 block basefee as a 64-bit integer (4 bytes)
+
+When included in the `transactions` array, this transaction should be
+RLP-encoded in the same way as other transactions.
+
+Here is an example valid `PayloadAttributesOPV1` object, which contains an L1
```

Comment:
Let's move this into a doc specific to the execution-engine?

Reply:
What about having the person that writes the spec move it?